### ISSUE TITLE
[bug]: some types are broken with TypeScript 4.8

### ISSUE THREAD

**Issue by bschlenk** [pmndrs/react-spring#1967]
### Which react-spring target are you using?

- [X] `@react-spring/web`
- [ ] `@react-spring/three`
- [ ] `@react-spring/native`
- [ ] `@react-spring/konva`
- [ ] `@react-spring/zdog`

### What version of react-spring are you using?

9.5.2

### What's Wrong?

TypeScript 4.8 doesn't like something about the `UseSpringProps` type and a few others.

```
node_modules/@react-spring/core/dist/declarations/src/hooks/useSpring.d.ts:8:141 - error TS2344: Type 'State' does not satisfy the constraint 'Lookup<any>'.               
                                                                                                                                                                           
8 export declare type UseSpringProps<Props extends object = any> = unknown & PickAnimated<Props> extends infer State ? Remap<ControllerUpdate<State> & {                   
                                                                                                                                              ~~~~~                        
                                                                                                                                                                           
  node_modules/@react-spring/core/dist/declarations/src/hooks/useSpring.d.ts:8:110                                                                                         
    8 export declare type UseSpringProps<Props extends object = any> = unknown & PickAnimated<Props> extends infer State ? Remap<ControllerUpdate<State> & {               
                                                                                                                   ~~~~~                                                   
    This type parameter might need an `extends Lookup<any>` constraint.
```

I looked into this a bit but react-spring's types are using a lot of advanced tricks I couldn't quite follow. This isn't super urgent because I can always add `skipLibCheck` to my project, mostly just wanted to make sure this was tracked somewhere.

### To Reproduce

Try to build a project with TypeScript 4.8 that uses the `useSpring` hook.

### Expected Behaviour

No typescript errors.

### Link to repo

https://github.com/bschlenk/react-spring-ts-48

**Comment by joshuaellis** [comment]
Thanks for reporting! Are you able to share a codesandbox that demonstrates the issue?

**Comment by bschlenk** [comment]
It only shows up when you run `tsc` so I wasn't sure how to trigger the error in codesandbox.

**Comment by joshuaellis** [comment]
I've started looking at this, but I need more headspace to figure out the rest of the errors, if you want to take a closer look check out this branch – https://github.com/pmndrs/react-spring/tree/fix/typescript-4.8
